### ISSUE TITLE
Open directly a haskell source file does not activate the extension

### ISSUE THREAD

**Issue by mbwgh** [alanz/vscode-hie-server#140]
I had a little trouble getting the plugin to work. Since I am new to VSCode as well, this is not necessarily a problem with `vscode-hie-server` per se, though.

After playing around with the editor, I came to the conclusion that not only do you have to be inside a stack or cabal project directory, but the directory has to be added to a VSCode Workspace as well. Only then will `hie`-related stuff happen, including output showing up in the console.

Is this assumption correct? Should this be documented more clearly?

**Comment by AlexeyRaga** [comment]
@mbwgh 

This extension hooks to VSCode's extensions API presenting itself as a language extension and implementing required API methods.
A documentation can be found here: https://code.visualstudio.com/api/language-extensions/overview

But basically, if you look inside `package.json` you'll find this:
```
  "activationEvents": ["onLanguage:haskell", "onLanguage:literate haskell"],
  "contributes": {
    "languages": [
      {
        "id": "haskell",
        "aliases": ["Haskell", "haskell"],
        "extensions": [".hs"]
      },
      {
        "id": "literate haskell",
        "aliases": ["Literate Haskell", "literate Haskell"],
        "extensions": [".lhs"]
      }
    ],
```

which, AFAIK, means that `VSCode` will call the extension's `activate` function (defined in `src/extension.ts`) wherever it detects that either `.hs` or `.lhs` file is opened.

The extension then would then subscribe to some events (like `workspace.onDidOpenTextDocument`) and communicate with HIE.

It is all in `src/extension.ts`, and VSCode docs could provide you with more context.

In my understanding, the extension is being developed by analogy to what already exists in other extensions, and what VSCode provides as examples, ad-hoq, so there is definitely a possibility that things can be done better. 
Feel free to suggest :)

**Comment by mbwgh** [comment]
@AlexeyRaga Sorry for taking so long to reply, thank you for your explanation.

I was able to test it a little more finally, and it seems that as you say, the `onLanguage` event is fired and trigger's the extension's activation. At least it says this in the `Log (Extension Host)`:

```
[2019-04-30 16:33:28.895] [exthost] [info] extension host started
...
[2019-04-30 16:33:29.830] [exthost] [info] ExtensionService#_doActivateExtension justusadam.language-haskell {"startup":false,"activationEvent":"onLanguage:haskell"}
[2019-04-30 16:33:29.830] [exthost] [info] ExtensionService#loadCommonJSModule /home/mbw/.vscode/extensions/justusadam.language-haskell-2.6.0/out/src/extension
[2019-04-30 16:33:29.832] [exthost] [info] eager extensions activated
[2019-04-30 16:33:29.886] [exthost] [info] ExtensionService#_doActivateExtension alanz.vscode-hie-server {"startup":false,"activationEvent":"onLanguage:haskell"}
[2019-04-30 16:33:29.886] [exthost] [info] ExtensionService#loadCommonJSModule /home/mbw/.vscode/extensions/alanz.vscode-hie-server-0.0.27/out/src/extension
```

This is from creating a new text file with the following content

```haskell
module Test where

import Data.Strin
```

so that I could test for GHC errors and autocompletion. Alas, even after closing and reopening vscode (according to the [documentation](https://code.visualstudio.com/api/references/activation-events#onLanguage), the event is fired on loading the file), nothing more seems to happen.

Only when I add the surrounding folder of the file to the workspace, there will be some output for `Haskell HIE (the-directory)` in the console, and visual highlighting of the error, and autocomplete seems to work as well, and the extension seems to know about only what is installed globally.

Is this the expected behavior?

**Comment by sheyll** [comment]
I have the exact same problem. When I open a Haskell source file in a folder with a stack project, vscode  used to start the plugin, now it doesn't.

I did not use vscode for a few months, they might have changed some stuff I might have missed, so I don't know if this is a problem I somehow caused.

But from a UX perspective: I didn't do anything bad, and it just doesn't work anymore ;)

**Comment by jneira** [comment]
I am able to reproduce this with latest versions of hie and the extension

**Comment by OctarineSorcerer** [comment]
I'm also able to reproduce this with the latest versions of both hie and extension

**Comment by bubba** [comment]
Fixed in #235
